src/rpc/network/communicator.h

```diff
@@ -28,11 +28,14 @@ class Sender {
   /*!
    * \brief Sender constructor
    * \param queue_size size (bytes) of message queue. 
+   * \param max_thread_count size of thread pool. 0 for no limit
    * Note that, the queue_size parameter is optional.
    */
-  explicit Sender(int64_t queue_size = 0) {
+  explicit Sender(int64_t queue_size = 0, int max_thread_count = 0) {
     CHECK_GE(queue_size, 0);
```

Comment:
Is there any check needed on `max_thread_count` when built without epoll support?

Reply:
no, Sender's thread limit just makes several senders to share one msg queue, not related to epoll. And receiver's thread limit will be handled in SocketReceiver::Wait